File: Samples/VB.Net/VBDotNet/IReportsDlg.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports SalesMatePlusLib

Namespace CSharpSDKTest
	Partial Public Class IReportsDlg
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Aditinalinfobtn_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles Aditinalinfobtn.CheckedChanged
			If Aditinalinfobtn.Checked = True Then
				HeadingTxt.Enabled = True
				AddHeadingTxt.Enabled = True
				CaptionTxt.Enabled = False
			End If
		End Sub

		Private Sub Withcaptionbtn_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles Withcaptionbtn.CheckedChanged
			If Withcaptionbtn.Checked = True Then
				HeadingTxt.Enabled = False
				AddHeadingTxt.Enabled = False
				CaptionTxt.Enabled = True
			End If
		End Sub

		Private Sub NormalBtn_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles NormalBtn.CheckedChanged
			If NormalBtn.Checked = True Then
				HeadingTxt.Enabled = False
				AddHeadingTxt.Enabled = False
				CaptionTxt.Enabled = False
			End If
		End Sub

		Private Sub AdTableBtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles AdTableBtn.Click
			Dim strAdTable As String
			strAdTable=Tablecombo.Text
			If strAdTable = "" Then
				MessageBox.Show("Enter The table name")
			Else
				Tablecombo.Items.Add(strAdTable)
				MessageBox.Show("Added Sucessfully")
				strAdTable = ""
			End If
		End Sub

		Private Sub Clsbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Clsbtn.Click
			Me.Close()
		End Sub

		Private Sub Reportbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Reportbtn.Click
			Dim m_Reports As IReports
			m_Reports = New ReportsClass()
			Dim strQuery As String
			strQuery=Tablecombo.Text
			If strQuery="" Then
				MessageBox.Show("Enter the table name")
			ElseIf NormalBtn.Checked = True Then
				m_Reports.FillReport(strQuery)
			ElseIf Withcaptionbtn.Checked = True Then
				Dim strCaption As String
				strCaption = CaptionTxt.Text
				If strCaption = "" Then
					MessageBox.Show("Please enter the caption")
				Else
					m_Reports.FillReportWithCaption(strQuery, strCaption)
				End If
			ElseIf Aditinalinfobtn.Checked = True Then
				Dim strHeading As String
				Dim strAdheading As String
				Dim bshowheader As Integer=1
				Dim bshowfooter As Integer=1
				strHeading=HeadingTxt.Text
				strAdheading=AddHeadingTxt.Text
				If strHeading = "" Then
					MessageBox.Show("Enter the Heading")
				ElseIf strAdheading = "" Then
					MessageBox.Show("Enter the additional Heading")
				ElseIf (strAdheading = "") AndAlso (strHeading = "") Then
					MessageBox.Show("Enter the Heading and Additional Heading")
				Else
					m_Reports.FillReportEx(strQuery, strHeading, strAdheading, bshowheader, bshowfooter)
				End If
			End If
		End Sub

	End Class
End Namespace

File: Samples/C#.Net/CSharpSDKTest/IReportsDlg.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using SalesMatePlusLib;

namespace CSharpSDKTest
{
    public partial class IReportsDlg : Form
    {
        public IReportsDlg()
        {
            InitializeComponent();
        }

        private void Aditinalinfobtn_CheckedChanged(object sender, EventArgs e)
        {
            if (Aditinalinfobtn.Checked == true)
            {
                HeadingTxt.Enabled = true;
                AddHeadingTxt.Enabled = true;
                CaptionTxt.Enabled = false;
            }
        }

        private void Withcaptionbtn_CheckedChanged(object sender, EventArgs e)
        {
            if (Withcaptionbtn.Checked == true)
            {
                HeadingTxt.Enabled = false;
                AddHeadingTxt.Enabled = false;
                CaptionTxt.Enabled = true;
            }
        }

        private void NormalBtn_CheckedChanged(object sender, EventArgs e)
        {
            if (NormalBtn.Checked == true) 
            {
                HeadingTxt.Enabled = false;
                AddHeadingTxt.Enabled = false;
                CaptionTxt.Enabled = false;
            }
        }

        private void AdTableBtn_Click(object sender, EventArgs e)
        {
            string strAdTable;
            strAdTable=Tablecombo.Text;
            if (strAdTable == "")
            {
                MessageBox.Show("Enter The table name");
            }
            else
            {
                Tablecombo.Items.Add(strAdTable);
                MessageBox.Show("Added Sucessfully");
                strAdTable = "";
            }
        }

        private void Clsbtn_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void Reportbtn_Click(object sender, EventArgs e)
        {
            IReports m_Reports;
            m_Reports = new ReportsClass();
            string strQuery;
            strQuery=Tablecombo.Text;
            if(strQuery=="")
            {
                MessageBox.Show("Enter the table name");
            }
            else if (NormalBtn.Checked == true)
            {
                m_Reports.FillReport(strQuery);
            }
            else if (Withcaptionbtn.Checked == true)
            {
                string strCaption;
                strCaption = CaptionTxt.Text;
                if (strCaption == "")
                {
                    MessageBox.Show("Please enter the caption");
                }
                else
                {
                    m_Reports.FillReportWithCaption(strQuery, strCaption);
                }
            }
            else if (Aditinalinfobtn.Checked == true)
            {
                string strHeading;
                string strAdheading;
                int bshowheader=1;
                int bshowfooter=1;
                strHeading=HeadingTxt.Text;
                strAdheading=AddHeadingTxt.Text;
                if (strHeading == "")
                {
                    MessageBox.Show("Enter the Heading");
                }
                else if (strAdheading == "")
                {
                    MessageBox.Show("Enter the additional Heading");
                }
                else if ((strAdheading == "") && (strHeading == ""))
                {
                    MessageBox.Show("Enter the Heading and Additional Heading");
                }
                else
                {
                    m_Reports.FillReportEx(strQuery, strHeading, strAdheading, bshowheader, bshowfooter);
                }
            }
        }

    }
}